enhancements/api-review/add-new-CRD-ImageContentPolicy(ICP)-to-config.openshift.io.md

```diff
@@ -0,0 +1,260 @@
+2. The rejection of conflicting `allowMirrorByTags` values:<br/>
+   The validation requires the `allowMirrorByTags` to be same for the same `source`. MCO will fail on the conflicting values.
+3. The merge order of mirrors for the same source:<br/>
+   Order is preserving the relative order of the mirrors, if possible.
+   For example,
+    - if policy A has mirrors `a, b, c` and policy B has mirrors `c, d, e`, the
+   mirrors will be used in the order `a, b, c, d, e`.
+   	- if policy A and policy B contain the mirrors cannot be merged by the relative order,
+	   they will be sorted in alphabetical order. If policy A has mirrors `a, d, e` and
+	   policy B has mirrors `b, c, e`, the `a, d` and `b, c` will be sorted in alphabetical order,
+	   mirrors will be used in the order `a, b, c, d, e`.
+    - if the orders of mirror entries conflict (e.g. `a, b` vs. `b, a`) the configuration is not
+	   rejected but the resulting order is unspecified.<br/>
+
+	If want to use a deterministic order of mirrors, configure them into one list of mirrors using expected order.
+4. The ImageStream has not supported pulling using tags yet. It has started working on this feature.
+
+### Risks and Mitigations
+
+Pulling images from mirror registries without the digest specifications could lead to returning different image version from different registry if the image tag mapping is out of sync. But the OpenShift core required image using digests to avoid different versions won't consume this feature at all, so it is not exposed to the risks that anyone who actually uses the feature will be exposed to.
+
+## Design Details
+
+### Test Plan
+
+Update the tests that are currently in the MCO to verify that `mirror-by-digest-only` not have been set when spec `allowMirrorByTags: true` is specified in the ImageContentPolicy.
+
+### Graduation Criteria
+
+GA in openshift will be looked at to
+determine graduation.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+- Enumerate service level indicators (SLIs), expose SLIs as metrics
+- Write symptoms-based alerts for the component(s)
+
+#### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+- Backhaul SLI telemetry
+- Document SLOs for the component
+- Conduct load testing
+
+#### Removing a deprecated feature
+
+- A Jira card [OCPNODE-717](https://issues.redhat.com/browse/OCPNODE-717) was created to record the upgrade of ImageContenPolicy(ICP) . The [repositories](https://docs.google.com/document/d/11FJPpIYAQLj5EcYiJtbi_bNkAcJa2hCLV63WvoDsrcQ/edit) currently rely on operator/v1alpha1 ImageContentSourcePolicy(ICSP) will be migrated to config/v1 ImageContentPolicy.
+
+### Upgrade / Downgrade Strategy
+
+Upgrades should not be affect. But the downgrade will be affected.
```

Comment:
On last week’s call we were talking about how this can’t be the first case of a schema migration, there must be a precedent and probably an established process for how to do it — but we didn’t know what exactly it was at that time.

Is there a known precedent and a mechanism that can be just followed?